data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/PeerForwarderProvider.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.peerforwarder;
+
+public class PeerForwarderProvider {
+    private final PeerForwarder peerForwarder;
+
+    public PeerForwarderProvider(final PeerForwarder peerForwarder) {
+        this.peerForwarder = peerForwarder;
+    }
+
+    public PeerForwarder register() {
+        // TODO: Refactor the AppConfig and PeerForwarder so that we only create these objects when necessary.
+        // TODO: Refactor PeerForward to make it an interface. Support both a configured peer-forward and a no-op local-only. Provide the one appropriate based on user configuration.
```

Comment:
I like the class design here. I'm not sure what a no-op would be here, would events still go through peer forwarder if local `PeerForwarder` is provided by `register()`?

Reply:
I'm actually thinking that we make `PeerForwarder` an interface. When there are no peers configured, the `PeerForwarderProvider` would provide a `LocalOnlyPeerForwarder` class which is set to return all input events.

e.g.

```
class LocalOnlyPeerForwarder implements PeerForwarder {
  public List<Record<Event>> forwardRecords(final Collection<Record<Event>> records,
                                              final Set<String> identificationKeys,
                                              final String pluginId) {
    return records;  // OK, we'd need to think through the List/Collection problem, but that can be overcome
  }
}
```